components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/attest_crypto.c

```diff
@@ -0,0 +1,285 @@
+    memcpy(buf_to_hold_y_coord.ptr,
+           (const void *)attest_key.puby_key,
+           attest_key.puby_key_size);
+
+    /* Update size */
+    buf_to_hold_y_coord.len = attest_key.puby_key_size;
+
+    x_coord->ptr = buf_to_hold_x_coord.ptr;
+    x_coord->len = buf_to_hold_x_coord.len;
+    y_coord->ptr = buf_to_hold_y_coord.ptr;
+    y_coord->len = buf_to_hold_y_coord.len;
+
+    return T_COSE_SUCCESS;
+}
+
+/**
+ * \brief Check some of the sizes for hash implementation.
+ *
+ * \return  Value from \ref t_cose_err_t error if sizes are not correct.
+ *
+ * It makes sure the constants in the header file match the local
+ * implementation.  This gets evaluated at compile time and will
+ * optimize out to nothing when all checks pass.
+ */
+static inline enum t_cose_err_t check_hash_sizes()
+{
+    if (T_COSE_CRYPTO_SHA256_SIZE != PSA_HASH_SIZE(PSA_ALG_SHA_256)) {
+        return T_COSE_ERR_HASH_GENERAL_FAIL;
+    }
+
+    return T_COSE_SUCCESS;
+}
+
+/**
+ * \brief Convert COSE algorithm ID to a PSA algorithm ID
+ *
+ * \param[in] cose_hash_alg_id   The COSE-based ID for the
+ *
+ * \return PSA-based hash algorithm ID, or MD4 in the case of error.
+ *
+ */
+static inline psa_algorithm_t cose_hash_alg_id_to_psa(int32_t cose_hash_alg_id)
+{
+    psa_algorithm_t return_value;
+
+    switch (cose_hash_alg_id) {
+        case COSE_ALG_SHA256_PROPRIETARY:
+            return_value = PSA_ALG_SHA_256;
+            break;
+        default:
+            return_value = PSA_ALG_MD4;
+            break;
+    }
+
+    return return_value;
+}
+
+enum t_cose_err_t
+t_cose_crypto_hash_start(struct t_cose_crypto_hash *hash_ctx,
+                         int32_t cose_hash_alg_id) {
+    enum t_cose_err_t cose_ret = T_COSE_SUCCESS;
+    psa_status_t psa_ret;
+    struct t_cose_psa_crypto_hash *psa_hash_ctx;
+
+    /* These next 3 lines optimize to nothing except when there is
+     * failure.
+     */
+    cose_ret = check_hash_sizes();
+    if (cose_ret)
+    {
+        return cose_ret;
+    }
+
+    psa_hash_ctx = (struct t_cose_psa_crypto_hash *)hash_ctx;
+
+    psa_ret = psa_hash_setup(&psa_hash_ctx->operation,
+                             cose_hash_alg_id_to_psa(cose_hash_alg_id));
+
+    if (psa_ret == PSA_SUCCESS)
+    {
+        psa_hash_ctx->status = PSA_SUCCESS;
+        cose_ret = T_COSE_SUCCESS;
+    } else if (psa_ret == PSA_ERROR_NOT_SUPPORTED)
+    {
+        cose_ret = T_COSE_ERR_UNSUPPORTED_HASH;
+    } else
+    {
+        cose_ret = T_COSE_ERR_HASH_GENERAL_FAIL;
+    }
+
+    return cose_ret;
+}
+
+void t_cose_crypto_hash_update(struct t_cose_crypto_hash *hash_ctx,
+                               struct useful_buf_c data_to_hash)
+{
+    struct t_cose_psa_crypto_hash *psa_hash_ctx;
+
+    psa_hash_ctx = (struct t_cose_psa_crypto_hash *)hash_ctx;
+
+    if (psa_hash_ctx->status == PSA_SUCCESS) {
+        if (data_to_hash.ptr != NULL) {
+            psa_hash_ctx->status = psa_hash_update(&psa_hash_ctx->operation,
```

Comment:
can update fail, should the return value be checked?

Reply:
if update fail you will return error while calling finish : 

https://github.com/ARMmbed/mbed-os/blob/1808bf26fe82033cf1134001debac8e203ed7cc4/components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/attest_crypto.c#L262-L282